models/repos.js

```diff
@@ -1,109 +1,205 @@
 const dialog = require('electron').remote.dialog
 const ipc = require('electron').ipcRenderer
+const waterfall = require('run-waterfall')
+const app = require('electron').remote.app
 const encoding = require('dat-encoding')
-const debounce = require('debounce')
+const Multidat = require('multidat')
+const toilet = require('toiletdb')
+const assert = require('assert')
+const mkdirp = require('mkdirp')
 const open = require('open')
-
-const Manager = require('../lib/dat-manager')
+const path = require('path')
 
 module.exports = createModel
 
-function createModel (cb) {
-  const manager = new Manager()
-
-  ipc.on('link', (ev, url) => {
-    const key = encoding.decode(url)
-    manager.download(key, (err) => {
-      if (err) throw err
-    })
-  })
-
-  ipc.send('ready')
-  ipc.on('log', (ev, str) => console.log(str))
+function createModel () {
+  var manager = null
 
   return {
     namespace: 'repos',
     state: {
-      updateIndex: 0,
+      downloadsDir: path.join(app.getPath('downloads'), '/dat'),
+      ready: false,
       values: []
     },
     subscriptions: {
-      manager: handleManagerUpdate
+      start: startMultidat,
+      onIpc: handleIpc
     },
     reducers: {
-      update: update
+      ready: multidatReady,
+      update: updateDats
     },
     effects: {
-      open: openDat,
-      share: shareDat,
       create: createDat,
-      delete: deleteDat,
-      download: downloadDat,
-      'add-dir': addDirectory,
-      deleteConfirm: deleteConfirmed
+      remove: removeDat,
+      open: openDirectory,
+      share: shareDat,
+      clone: cloneDat
     }
   }
 
-  function handleManagerUpdate (send, done) {
-    manager.on('update', debounce(() => {
-      send('repos:update', manager.get(), done)
-    }), 1500, true)
+  // boot multidat, create the ~/Downloads/dat directory
+  function startMultidat (send, done) {
+    var dblocation = path.join(process.env.HOME)
+    var dbfile = path.join(dblocation, '.dat-desktop-db.json')
+
+    var arr = [
+      function (next) {
+        mkdirp(dblocation, next)
+      },
+      function (val, next) {
```

Comment:
we're not using `val`, can you rename it to `_`?
also the `val` below would be more clear when renamed to `multidat`, as that's the identifier used below.